RSSReader/Presenter/FeedPresenter.m

```diff
@@ -41,12 +41,13 @@ - (void)asyncLoadNewsWithCompletion:(void (^)(NSError *))completion {
 
 - (void)loadNews {
     @autoreleasepool {
+        __block typeof(self) weakSelf = self;
```

Comment:
because Im using self inside the body of the block. Why not?

Reply:
Because __block modifier manages the way local variables are cached by the block. As far as I can see you change no values outside of the block scope. Am I right?

Reply:
If we use self inside the block, then we should weakify the reference to it to avoid retain cycle